GA4 client_id (#996)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx

| Parameters    | Mapping value  |
| :-------------| :-----------------|
| `client_id`         |  Refer to the <Link to="#mapping-client_id">Mapping `client_id`</Link> section. |
| `app_instance_id`   |  from `externalID` `ga4AppInstanceId`  | 

<div class="infoBlock">
Refer to the <Link to="#how-do-i-obtain-app_instance_id">FAQ</Link> section for more information on how to obtain the <code class="inline-code">app_instance_id</code>.
</div>

### Mapping `client_id` 

> RudderStack maps the `client_id` in the following order of precedence:

Reviewer: ```suggestion
RudderStack maps the GA4 `client_id` in the following order of precedence:
```

> - If a value is provided in the RudderStack dashboard's **GA4 client_id Field Identifier** setting, it looks up that value in the event payload and maps it to the GA4 `client_id`.

Reviewer: ```suggestion
- If a value is provided in the RudderStack dashboard's **GA4 client_id Field Identifier** setting, it looks up that value in the event payload and maps it to `client_id`.
```

> - If no value is provided in the above dashboard setting or a value is provided in the dashboard but not in the payload, RudderStack maps the `client_id` from the following fields in the same priority order as listed below:

Reviewer: ```suggestion
- If no value is provided in the above dashboard setting or a value is provided in the dashboard but not in the event payload, then RudderStack maps `client_id` from the following fields in the exact priority order as listed below:
```

  - From `externalID` `ga4ClientId`
  - From `externalID` `anonymousId`

>   - From `messageId`

Reviewer: Wondering if it we should present this as 
`externalId: ga4ClientId`
`externalId: anonymousId` 

as `externalId` is an object within which `ga4ClientId` and `anonymousId` are the fields. 

Please correct me if I'm wrong here. @ishachopra0401 @mihir-4116

Reviewer 2: no,`anonymousId`  will not come from externalId, just `anonymousId` is enough

| Parameters    | Mapping value  |
| :-------------| :-----------------|
| `client_id`         |  Refer to the <Link to="#mapping-client_id">Mapping `client_id`</Link> section. |
| `app_instance_id`   |  from `externalID` `ga4AppInstanceId`  | 

<div class="infoBlock">
Refer to the <Link to="#how-do-i-obtain-app_instance_id">FAQ</Link> section for more information on how to obtain the <code class="inline-code">app_instance_id</code>.
</div>

### Mapping `client_id` 

RudderStack maps the `client_id` in the following order of precedence:

- If a value is provided in the RudderStack dashboard's **GA4 client_id Field Identifier** setting, it looks up that value in the event payload and maps it to the GA4 `client_id`.
- If no value is provided in the above dashboard setting or a value is provided in the dashboard but not in the payload, RudderStack maps the `client_id` from the following fields in the same priority order as listed below:
  - From `externalID` `ga4ClientId`
  - From `externalID` `anonymousId`

>   - From  `context.device.id`

Reviewer 2: Remove - From `context.device.id`
this is not required now
